code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/builders/AepPushNotificationBuilder.kt

```diff
@@ -0,0 +1,971 @@
+     * @param customSound `String` containing the custom sound to apply on the notification channel
+     * @param importance [Int] containing the notification importance
+     * @return A [String] containing the created or existing channel ID
+     */
+    internal fun createChannel(
+        context: Context,
+        channelId: String?,
+        customSound: String?,
+        importance: Int
+    ): String {
+        // No channel creation required.
+        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+            return channelId ?: PushTemplateConstants.DEFAULT_CHANNEL_ID
+        }
+
+        // Decide on which channel to use provided or default
+        val notificationManager =
+            context.getSystemService(Context.NOTIFICATION_SERVICE) as NotificationManager
+        val channelIdToUse = channelId ?: PushTemplateConstants.DEFAULT_CHANNEL_ID
+
+        // create the channel
+        createNotificationChannel(
+            context,
+            notificationManager,
+            channelIdToUse,
+            customSound,
+            importance
+        )
+        // return the id of the channel.
+        return channelIdToUse
+    }
+
+    @RequiresApi(Build.VERSION_CODES.O)
+    private fun createNotificationChannel(
+        context: Context,
+        notificationManager: NotificationManager,
+        channelId: String,
+        customSound: String?,
+        importance: Int
+    ) {
+        // Don't create a channel if it already exists
+        if (notificationManager.getNotificationChannel(channelId) != null) {
+            Log.trace(
+                PushTemplateConstants.LOG_TAG,
+                SELF_TAG,
+                "Using previously created notification channel: $channelId."
+            )
+            return
+        }
+
+        // Create a channel
+        val channel = NotificationChannel(
+            channelId, PushTemplateConstants.DEFAULT_CHANNEL_NAME, importance
+        )
+
+        // Add a sound if required.
+        val sound = if ((customSound.isNullOrEmpty())) {
+            RingtoneManager.getDefaultUri(RingtoneManager.TYPE_NOTIFICATION)
+        } else getSoundUriForResourceName(customSound, context)
+        channel.setSound(sound, null)
+
+        Log.trace(
+            PushTemplateConstants.LOG_TAG,
+            SELF_TAG,
+            "Creating a new notification channel with ID: $channelId. ${if (customSound.isNullOrEmpty()) "and default sound." else "and custom sound: $customSound."}"
+        )
+        notificationManager.createNotificationChannel(channel)
+    }
+
+    /**
+     * Creates a silent notification channel if the device is running on Android O or higher. If the
+     * channel already exists, the same channel is used.
+     *
+     * @param notificationManager the [NotificationManager]
+     * @param importance [Int] containing the notification importance
+     */
+    @RequiresApi(Build.VERSION_CODES.O)
+    internal fun setupSilentNotificationChannel(
```

Comment:
Can we achieve the same with `createNotificationChannel` method in this class? We can modify `createNotificationChannel` to also accept channel name and call like so

```
createNotificationChannel(
        context,
        PushTemplateConstants.DefaultValues.SILENT_NOTIFICATION_CHANNEL_ID,
        PushTemplateConstants.SILENT_CHANNEL_NAME,
        PushTemplateConstants.DefaultValues.SILENT_SOUND,
        importance
)
```

Reply:
yes, a common channel creation method can be used if given the parameters to use. i'll update.

Reply:
Will address in subsequent PR when moving it to extensions